### DISCUSSION TITLE
Wish there is a book on Avalonia

### DISCUSSION THREAD

**Discussion by WarriorBlue** [AvaloniaUI/Avalonia#14798] [Ideas]
Avalonia is a powerful and amazing framework. Is there any plan to write a book on it? As a hobbyist, I cannot master it only by self-study or the articles on the Analonia website, this is upset. Maybe for a lot of hobbyists,  this is a big challenge to master it.
Thank you very much.

**Comment by thevortexcloud** [comment]
It's fairly similar to WPF. While there are differences, a lot of what works in WPF will work in Avalonia with only minor changes. So while I am not sure if there is an Avalonia book, there are certainly WPF ones you can use as a starting point.

**Comment by stevemonaco** [comment]
The Avalonia community isn't big enough to commercially support the writing of a book which would take several months, still not cover the needs of everybody, and many will complain it's not video content. Several WPF books of the era were well over a thousand pages which nobody wants to read as a beginner these days. You can't master any programming concept by reading, you need sizable amounts of practice to clear up misconceptions and experiment.

What's more valuable is building a core set of Avalonia fundamentals and a sense of what exists in the ecosystem so that you can look it up when you need it. When a book goes to press, it's already outdated.

If there are particular concepts that are hard to understand (either in existing docs or they don't exist at all), then create an issue in https://github.com/AvaloniaUI/avalonia-docs

**Comment by MikeCodesDotNET** [comment]
I understand why you'd like a book about Avalonia, but it's likely to come from somewhere other than the core team. Let me explain why:

**Time vs. Reward**
Several publishers have asked us to write for them, but the pay is low for the massive effort involved. We'd rather spend that time improving Avalonia itself, which benefits everyone. 

**Accessibility** 
Fundamental learning resources should be free for everyone. Books cost money, but our online documentation is freely accessible.

**Staying Current**
Tech evolves FAST. Books become outdated quickly, and we'd rather focus on resources that stay up-to-date with Avalonia's progress. My stack of outdated Xamarin books proves the point!

**To any budding authors** 
With all that said, if any budding authors in the community want to write a book on Avalonia, we'd be delighted. Writing a book is a great way to increase credibility and open new doors. It's often a stepping stone to getting speaking slots at conferences and promotions. 

We ARE committed to making Avalonia easier to learn. That means fantastic documentation, helpful tutorials, and a supportive community. 

Let me know what specific resources would be most beneficial for you.

**Comment by WarriorBlue** [comment]
Thank you for the replies, I am appreciated. I know the team's concerns, and I will study it with my full efforts, wish that one day I can  make my contribution to the project.
Thank you again.

**Reply by timunie** [reply]
Hey @WarriorBlue ,

if you need some guidance feel free to join https://t.me/Avalonia . If you want to contribute, let us know which docs where probably unclear or missing for you. If these are simple to fix, you are also welcome to file a PR there.

**Reply by WarriorBlue** [reply]
OK. I got it. That's very kind of you

**Comment by reelthymeos** [comment]
Hi everyone,

I will be building a simple desktop app in C# for my company. The purpose of this app will be interfacing with and controlling a piece of test & measurement equipment. After careful consideration, I believe that the two most reasonable routes for me are WPF or Avalonia. Having a solution that is cross platform is a "nice to have" but by no means a hard requirement. The more important thing is that I get an app up and running in relatively short order. 

I am somewhat familiar with WPF and XAML. Ideally I would like to go the Avalonia route, since it looks like the future and WPF is a legacy platform. However, my main concern is the relative lack of documentation/tutorials/forums for Avalonia when compared with WPF, which has extensive materials available. I just don't want to bite off more than I can chew, especially considering the aggressive schedule for this app. 

Can anyone offer some advice or insight into the situation with regards to Avalonia literature? What has your experience been? Do you think my concern is warranted?

**Reply by thevortexcloud** [reply]
The official tutorials are pretty decent. I personally used the music store tutorial to learn it. The documentation is however lacking in some areas, but it's far better than it used to be. I still sometimes however end up searching for WPF tutorials/examples as most the time they work in Avalonia with some tweaks. As such if you already know WPF then I would argue there is not much of a learning curve. There is also fairly decent community support here on github and you can generally get a response fairly quickly for most problems.

The time to actually develop something is also pretty fast too. But that's mainly because of the whole binding system, which WPF has also.

**Reply by timunie** [reply]
- check also Avalonia.Samples
- Join us on Telegram https://t.me/Avalonia
- If you need better support, we offer paid support as well

**Comment by LaurentInSeattle** [comment]
Writing a book is a lot of work... And that work does not pay well. :( 

Of note, most WPF books were written by Microsoft employees, cashing their regular paychecks. And then published by Microsoft press. In other words, readers were paying for somewhat "advanced" documentation.

Nowadays, still the same, but no more books, videos on YouTube instead.

**Comment by vessillo** [comment]
Hi everyone,
For the purpose of teaching multiplatoform programming in C#, I have written a book on AvaloniaUI in order to be able to develop a desktop application using MVVM pattern.
The ebook is written in italian language.
The table of contents is the following

![image](https://github.com/user-attachments/assets/15128305-9ea0-42c4-936e-6ed11a5fd2de)

the techniques used are the following 

![image](https://github.com/user-attachments/assets/c0efadcc-da47-4ff5-bb4c-4ace66984542)

The bookshelf is here

https://www.lulu.com/shop/roberto-bandiera/avaloniaui/ebook/product-65rq4kp.html

Hope it helps.
   Roberto

**Reply by LaurentInSeattle** [reply]
Molto probabilmente si tratta del primo libro scritto per Avalonia, congratulazioni per essere il primo!

**Reply by lizanx** [reply]
Really hope it's written in English so that it can benefit many more learners.

**Reply by WarriorBlue** [reply]
same as lizanx, wish there is a english version for different learners.

**Reply by vessillo** [reply]
I'm going to manage the translation in english

**Reply by lizanx** [reply]
> I'm going to manage the translation in english

Would like to be notified when it comes out!

**Reply by WarriorBlue** [reply]
@vessillo is there any progress of the book? 
thank you

**Reply by vessillo** [reply]
The ebook in english language has just been published.
Find it at lulu.com (epub) and amazon.com (kindle version).
The paper version needs some days for revising the draft.
 Roberto

**Reply by michael-mmt** [reply]
Hi, @vessillo, I would like to translate your book into Chinese and publish it in China. Is this possible? How can I contact you? My email address is 311155@qq.com

**Reply by xflcx1991** [reply]
> Hi, @vessillo, I would like to translate your book into Chinese and publish it in China. Is this possible? How can I contact you? My email address is [311155@qq.com](mailto:311155@qq.com)您好，我想将您的书籍翻译成中文并在中国出版。请问这可行吗？我该如何联系您？我的邮箱地址是 311155@qq.com。

Greet！

**Comment by vessillo** [comment]
I am adding 2 more chapter to the first edition.I hope to finish it in few days.Roberto Inviato dal mio Galaxy
-------- Messaggio originale --------Da: WarriorBlue ***@***.***> Data: 18/05/25  09:54  (GMT+01:00) A: AvaloniaUI/Avalonia ***@***.***> Cc: vessillo ***@***.***>, Mention ***@***.***> Oggetto: Re: [AvaloniaUI/Avalonia] Wish there is a book on Avalonia (Discussion #14798) 
@vessillo is there any progress of the book?
thank you

**Reply by WarriorBlue** [reply]
@vessillo that is wonderful, the first avalonia book in the world.

**Reply by vessillo** [reply]
Today the paper version in English and Italian was also released

**Comment by lizanx** [comment]
Glad that the English version is available now!
Will definitely read it when I loosen my hands from recent work.

I'm posting the link to the book here for everyone's convenience: [AvaloniaUI Cross-platform Desktop Applications in C#](https://www.lulu.com/shop/roberto-bandiera/avaloniaui/ebook/product-2mwqqr2.html)
